functions/go/set-image/image_transformer.go

```diff
@@ -39,6 +42,57 @@ type SetImage struct {
 	Image types.Image `json:"image,omitempty" yaml:"image,omitempty"`
 	// AdditionalImageFields is used to specify additional fields to set image.
 	AdditionalImageFields types.FsSlice `json:"additionalImageFields,omitempty" yaml:"additionalImageFields,omitempty"`
+	// setImageResults is used internally to track which images were updated
+	setImageResults setImageResults
+}
+
+// setImageResultKey is used as a unique identifier for set image results
+type setImageResultKey struct {
+	ResourceRef yaml.ResourceIdentifier
+	// FilePath is the file path of the resource
+	FilePath string
+	// FileIndex is the file index of the resource
+	FileIndex int
+	// FieldPath is field path of the image field
+	FieldPath string
+}
+
+// setImageResult maps a previous image value to a new image value where set-image is applied
+// e.g. "nginx:1.20.2" -> "nginx:1.21.6"
+type setImageResult struct {
+	// CurrentValue is the value before applying the set-image mutation
+	CurrentValue string
+	// ProposedValue is the value that will be set after applying set-image
+	ProposedValue string
```

Comment:
This name doesn't feel very accurate. Maybe call it `UpdatedValue`.

Reply:
I based this on the [analogous type from the SDK](https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/thirdparty/kyaml/fnsdk/result.go#L111). Do you feel that this has a different context?

Reply:
SGTM, done